### PR TITLE
Fixed FeatherGridContextMenu placement in jupyter notebook and voila

### PR DESCRIPTION
**Pull request by alphaparrot**
*Issue number of the reported bug or feature request: #422*

**Describe your changes**
Added a section of code that repositions the FeatherGridContextMenu widget to the actual click location, using the geometry of the page and document body. This fixes the bug in [https://github.com/bloomberg/ipydatagrid/issues/422](https://github.com/bloomberg/ipydatagrid/issues/422) (and also the original bug, documented in [https://github.com/voila-dashboards/voila/issues/930](https://github.com/voila-dashboards/voila/issues/930)), which is caused by changes in Lumino originally proposed in [https://github.com/jupyterlab/lumino/issues/329#issuecomment-1266200387](https://github.com/jupyterlab/lumino/issues/329#issuecomment-1266200387), along with the retention of absolute positioning for this element in ipydatagrid. 

**Testing performed**
Tested in two different notebooks (one bare-bones/MWE and one full-featured from a production setting) in jupyter notebook and voila (see attached image for example from barebones notebook). If page content changes immediately as a result of opening the filter dialog (such as an output widget getting cleared), this will cause the context menu to appear out of position. This could potentially be remedied by repositioning the menu a _second_ time after the menu is fully opened, but this is likely to be excessive and not a good solution to that particular problem. 

image

In the future, a departure from absolute positioning may be required, which may obviate the need for a repositioning operation here.

### CODE REVIEW

**Review by martinRenou — COMMENTED** [review]
Thank you! This is indeed not great to have to workaround this behavior, we may still want to open an issue to upstream lumino to track this

**Comment by martinRenou** [comment]
It is required that you signoff your commits before we can merge.

To signoff your latest commit:
```bash
git commit --amend --signoff 
```

**Comment by alphaparrot** [comment]
Whoops, sorry about that. I think that should be fixed now.

**Review by gaborbernat — APPROVED** [review]

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**
